Simplify condition check in kuttl tests

kajinamit opened the pull request:
Currently we assert every single field in conditions but this requires updating the expected values in case any message is updated in the underlying dependencies.

Because we have env test which is more suitable to assert conditions, this simplifies the assertions in kuttl tests.

gibizer commented:
We need https://github.com/openstack-k8s-operators/lib-common/pull/26 to lend first, until that this will be red

stuggi commented:
@gibizer sorry, somehow missed that, could you rebase this. we can then land it

gibizer commented:
Rebased. Dependencies has been merged so this is good to go.

stuggi reviewed (approved):
/lgtm

kajinamit committed:
Simplify condition check in kuttl tests

Currently we assert every single field in conditions but this requires
updating the expected values in case any message is updated in
the underlying dependencies.

Because we have env test which is more suitable to assert conditions,
this simplifies the assertions in kuttl tests.

gibizer commented:
> Currently we assert every single field in conditions but this requires updating the expected values in case any message is updated in the underlying dependencies.
> 
> Because we have env test which is more suitable to assert conditions, this simplifies the assertions in kuttl tests.

I don't think we have envtest coverage in placement for all the conditions. I think that needs to be added first.

kajinamit commented:
Let me reconsider the approach. The initial motivation of making this is avoid baking all condition message which requires us to update kuttl test case when any change is made in underlying modules. envtest would work better for that point(as we can import message from the underling lib) but for that we need to make sure we have all test cases to cover all conditions.

In addition current kuttl does not allow asserting only the first element of the conditions list, which would require a separate step to query output but that might not be a good way.
